src/site/content/en/blog/form-fatigue-validate-user-input/index.md

```diff
@@ -0,0 +1,308 @@
+---
+title: "Reduce form fatigue: validate user input"
+author: mdiblasio
+date: 2019-05-02
+description: |
+  This post describes how to validate user input.
+wf_blink_components: Blink>Accessibility
+tags:
+  - post
+  - accessibility
+  - ux
+---
+
+_This is post 4 of 5 of the [Reduce form fatigue series](../form-fatigue) that
+shows you how to build better forms for online stores._
+
+<!-- TODO (robdodson): is there a format we should use for this line that opens each article?  -->
+
+<!-- TODO (dutton): is there a format we should use for this line that opens each article?  -->
+
+No matter how optimized your forms are, users will make mistakes! When this
+happens, provide them with helpful feedback in real-time so they can quickly fix
+the issue. 
+
+## Restrict user input
+
+An effective way to avoid validation errors is to restrict user input (where
+possible) using built-in attributes.
+
++   **For numeric input**, use the `min` and `max` attributes to limit
+    input within a range. The `step` attribute can specify how much input
+    should each increment/decrement when adjusted by the slider or spinners.
+    For example, you can restrict a maximum price filter within a range of $50
+    to $500 with a `step` of 50:
+
+```html
+<input type="range" min="50" max="500" step="50">  
+```
+
+<!-- TODO (meggin):  image illustration or inline glitch? -->
+
++   **For text input**, use the `minlength` and `maxlength` attributes to
+    enforce limits on the length of text. For example, you can limit a
+    customized message on a flower delivery card to 50 characters:
+
+```html
+<input type="text" maxlength="50" >  
+```
+
+{% Aside 'caution' %}
+`minlength` and `maxlength` aren't enforced on mobile browsers. Always
+manually validate input on form submission.
+{% endAside %}
+
++   **For date and time input**, use `min` and `max` to restrict dates
+    and times. Browsers that provide a UI for choosing dates may prevent users
+    from entering invalid dates. For example, prevent users from selecting a
+    delivery date in the past or a delivery time outside of your operating hours:
+
+<figure class="w-figure">
+  <img src="image-browser-date-selector.png" alt="browser built-in date selector" style="max-width: 400px;">
```

Comment:
The text at the bottom of this image seems pretty compressed. Possible to swap out with a higher quality version?

I'd also suggest adding `w-screenshot` to the image and centering it. Anytime the image doesn't provide its own border, `w-screenshot` can be used to make it so the image doesn't look like it's floating on the page.